command/agent/command.go

```diff
@@ -295,13 +295,26 @@ func (c *Command) setupAgent(config *Config, logOutput io.Writer, logWriter *log
 		return err
 	}
 
-	rpcListener, err := net.Listen("tcp", rpcAddr.String())
+	if _, ok := rpcAddr.(*net.UnixAddr); ok {
+		// Remove the socket if it exists, or we'll get a bind error
+		_ = os.Remove(rpcAddr.String())
```

Comment:
I actually am purposefully not checking for the error. If the socket doesn't already exist, Remove will error, but in that case we don't care. If the socket does exist and cannot be bound to, we'll error out on the next call (net.Listen) anyways.

I can remove the `_=` if you like but I tend to be explicit about such things for two reasons: one, to make it clear that there *is* a return value but we are choosing to ignore it; two, because in some cases not doing with some standard library stuff can cause a call to panic rather than providing an error, and as I tend to forget which cases those are, I try to avoid them altogether :-)